server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WhiteListManager.java

```diff
@@ -244,10 +244,10 @@ public void init() throws MessagingException {
     public void service(Mail mail) throws MessagingException {
 
         // check if it's a local sender
-        MailAddress senderMailAddress = mail.getSender();
-        if (senderMailAddress == null) {
+        if (mail.hasSender()) {
```

Comment:
did you inversed the condition? No test on this?

Reply:
Good catch but... This class was not tested at all.